qcelemental/models/results.py

```diff
@@ -68,21 +47,64 @@ class ResultInput(BaseModel):
     molecule: Molecule
     driver: DriverEnum
     model: Model
-    schema_name: constr(strip_whitespace=True, regex='qc_schema_input') = "qc_schema_input"  # IDEs complain, its fine
+    schema_name: constr(strip_whitespace=True,
+                        regex=qcschema_input_default) = qcschema_input_default
     schema_version: int = 1
     keywords: dict = {}
     provenance: Provenance = provenance_stamp(__name__)
 
     class Config:
         allow_mutation = False
+        allow_extra = True
+
+
+class FailedResult(ResultInput):
```

Comment:
I think these Failed classes are a bit much. Best to have a single `ComputeError` class where we have:
```python
success = False
input_data = Any
error_type = str
error_message = str
```